exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/FilePushDownFilter.java

```diff
@@ -244,4 +247,15 @@ protected void doOnMatch(RelOptRuleCall call, FilterPrel filter, ProjectPrel pro
     final RelNode newFilter = filter.copy(filter.getTraitSet(), Collections.singletonList(newNode));
     call.transformTo(newFilter);
   }
+
+  /**
+   * Checks whether specified group scan supports filter push down.
+   *
+   * @param groupScan group scan to check
+   * @return {@code true} if specified group scan supports filter push down, {@code false} otherwise
+   */
+  private static boolean supportsFilterPushDown(GroupScan groupScan) {
+    return groupScan instanceof AbstractGroupScanWithMetadata
```

Comment:
This kind of logic is fragile. Can we just add a method on the group scan that will answer this question, then let each group scan figure out the answer or itself?